src/imports/background/import-item-creation.js

```diff
@@ -0,0 +1,195 @@
+
+                // Filter out items already existing in DB, if specified
+                if (alreadyExists == null || !alreadyExists(encodedUrl)) {
+                    accItems.set(encodedUrl, transform(currItem))
+                }
+                return accItems
+            } catch (error) {
+                // URI malformed; don't add to list
+                return accItems
+            }
+        }, new Map())
+}
+
+/**
+ * @param {number} [maxResults=9999999] The max amount of items to grab from browser.
+ * @param {number} [startTime=0] The time to start search from for browser history.
+ * @returns {Array<browser.history.HistoryItem>} All history items in browser filtered by URL.
+ */
+const getHistoryItems = (maxResults = 9999999, startTime = 0) =>
+    browser.history.search({
+        text: '',
+        maxResults,
+        startTime,
+    })
+
+/**
+ * @param {number} [maxResults=100000] The max amount of items to grab from browser.
+ * @returns {Array<browser.bookmarks.BookmarkTreeNode>} All bookmark items in browser filtered by URL.
+ */
+const getBookmarkItems = (maxResults = 100000) =>
+    browser.bookmarks.getRecent(maxResults)
+
+export async function getOldExtItems() {
+    let bookmarkUrls = new Set()
+    const {
+        [OLD_EXT_KEYS.INDEX]: index,
+        [OLD_EXT_KEYS.BOOKMARKS]: bookmarks,
+    } = await browser.storage.local.get({
+        [OLD_EXT_KEYS.INDEX]: { index: [] },
+        [OLD_EXT_KEYS.BOOKMARKS]: '[]',
+    })
+
+    if (typeof bookmarks === 'string') {
+        try {
+            bookmarkUrls = new Set(JSON.parse(bookmarks).map(bm => bm.url))
+        } catch (error) {}
+    }
+
+    const transform = transformOldExtDataToImportItem(bookmarkUrls)
+    const importItems = []
+
+    // Only attempt page data conversion if index + bookmark storage values are correct types
+    if (index && index.index instanceof Array) {
+        // NOTE: There is a bug with old ext index sometimes repeating the index keys
+        //  (eg.I indexed 400 pages, index contained 43 million)
+        const indexSet = new Set(index.index)
+
+        for (const keyChunk of chunk(200)([...indexSet])) {
+            const storageChunk = await browser.storage.local.get(keyChunk)
+
+            keyChunk.forEach((key, i) => {
+                if (storageChunk[key] == null) {
+                    return
+                }
+
+                importItems.push(transform(storageChunk[key], i))
+            })
+        }
+    }
+
+    return importItems
+}
+
+/**
+ * @param {any} allowTypes
+ * @returns {any} Object containing three Maps of encoded URL keys to import item values.
+ *   Used to create the imports list and estimate counts.
+ */
+export default async function createImportItems(allowTypes = DEF_ALLOW) {
+    const filterItemsByUrl = await initFilterItemsByUrl()
+
+    // Get all history from browser, filter on existing DB pages
+    let historyItemsMap = filterItemsByUrl(
+        await getHistoryItems(),
```

Comment:
if you have 2 await is probably better to use parallel promises (if this function is not called within the loop it's not very relevant)